src/grpc/dp_grpc_conv.cpp

```diff
@@ -203,19 +245,13 @@ void DpToGrpcFwrule(const struct dp_fwall_rule *dp_rule, FirewallRule *grpc_rule
 		grpc_rule->set_action(FirewallAction::DROP);
 
 	src_ip = new IpAddress();
-	src_ip->set_ipver(IpVersion::IPV4);
-	src_ip->set_address(GrpcConv::Ipv4ToStr(dp_rule->src_ip));
 	src_pfx = new Prefix();
-	src_pfx->set_allocated_ip(src_ip);
-	src_pfx->set_length(__builtin_popcount(dp_rule->src_ip_mask));
+	SetupIpAndPrefix(dp_rule, src_ip, src_pfx, is_src);
 	grpc_rule->set_allocated_source_prefix(src_pfx);
 
 	dst_ip = new IpAddress();
-	dst_ip->set_ipver(IpVersion::IPV4);
-	dst_ip->set_address(GrpcConv::Ipv4ToStr(dp_rule->dest_ip));
 	dst_pfx = new Prefix();
-	dst_pfx->set_allocated_ip(dst_ip);
-	dst_pfx->set_length(__builtin_popcount(dp_rule->dest_ip_mask));
+	SetupIpAndPrefix(dp_rule, dst_ip, dst_pfx, !is_src);
```

Comment:
I like that this is more readable than using `true` and `false`, it just tripped me up that it's a variable. Usually I see this done as `#define IS_SRC true` and `#define IS_DST false`, since uppercase does not indicate a variable

Reply:
I am always skeptical using such #defines in c++ context. Defines are for me too much c-like :)  
I am opting for constexpr then smt like this:

>  constexpr bool IS_SRC = true;
     constexpr bool IS_DST = false;

Reply:
Perfect. That is both C++ like and it solves the variable problem. I need to remember this for later use.